lib/unittest.bzl

```diff
@@ -75,6 +77,13 @@ def _make(impl, attrs = None):
         attrs = attrs,
         _skylark_testable = True,
         test = True,
+        # is_host_windows uses a select(), therefore we cannot pass an output function that would
+        # compute the output file's name based on this attribute (to return "%{name}.bat" or
+        # "%{name}), so we use "%{name}.bat" on all platforms.
+        # On Windows this allows the output file to be executable. On all other platforms the file's
+        # extension is irrelevant because it's shebang line defines the interpreter.
+        outputs = {"testbin": "%{name}.bat"},
```

Comment:
Can you avoid using `outputs =`?
Instead, declare a file (`ctx.actions.declare_file`) in the implementation function. As a bonus, it won't need the .bat extension on non-Windows platforms.